### ISSUE TITLE
TypeError in downloading and parsing the lightning data 

### ISSUE THREAD

**Issue by jeppoo1** [pnuu/fmiopendata#6]
Hello,
I'm getting the following error `TypeError: a bytes-like object is required, not 'NoneType'` when trying to execute the script for downloading and parsing the lightning data, straight from the Github page, **in Azure Databricks** 
(Cluster Mode Standard
Databricks Runtime Version 6.5 
6.5 ML (includes Apache Spark 2.4.5, Scala 2.11)
Python Version 3).

This script:
`import fmiopendata`

`from fmiopendata.wfs import download_stored_query`
`lightning1 = download_stored_query("fmi::observations::lightning::multipointcoverage")`

`lightning1.latitudes  # Latitude of the lightning event [° North]`
`lightning1.longitudes  # Longitude of the lightning event [° East]`
`lightning1.times  # Time of the lightning event [datetime]`
`lightning1.cloud_indicator  # Indicator for cloud flashes (1 == cloud lightning)`
`lightning1.multiplicity  # Multiplicity of the lightning event`
`lightning1.peak_current  # Maximum current of the lightning event [kA]`
`lightning1.ellipse_major  # Location accuracy of the lightning event [km]`

returns the following error:

_TypeError: a bytes-like object is required, not 'NoneType'_

_TypeError                                 Traceback (most recent call last)_
_<command-1027777503111693> in <module>_
      _4 # Lightning data_
      _5 from fmiopendata.wfs import download_stored_query_
_----> 6 lightning1 = download_stored_query("fmi::observations::lightning::multipointcoverage")_
      _7_ 
      _8_ 

_/databricks/python/lib/python3.7/site-packages/fmiopendata/wfs.py in download_stored_query(query_id, args)_
    _124         raise NotImplementedError("No parser available for %s" % query_id)_
    _125_ 
_--> 126     return download_and_parse(query_id, args=args)_

_/databricks/python/lib/python3.7/site-packages/fmiopendata/lightning.py in download_and_parse(query_id, args)_
    _110     xml = read_url(url)_
    _111     mode = query_id.split("::")[-1]_
_--> 112     return Lightning(xml, mode)_

_/databricks/python/lib/python3.7/site-packages/fmiopendata/lightning.py in __init__(self, xml, mode)_
      47             self._parse_simple()_
     48         elif mode == "multipointcoverage":_
_---> 49             self._parse_multipoint()_
     _50         else:_
     _51             raise NotImplementedError("No parser for %s" % mode)_

_/databricks/python/lib/python3.7/site-packages/fmiopendata/lightning.py in _parse_multipoint(self)_
     _90_ 
     _91         """_
_---> 92         positions = np.fromstring(self._xml.findtext(wfs.GMLCOV_POSITIONS), dtype=float, sep=" ")_
     _93         self.latitudes = positions[::3]_
     _94         self.longitudes = positions[1::3]_

_TypeError: a bytes-like object is required, not 'NoneType'__

If I try to run the "simple" version of the lightning script, the following code only returns empty dataframes:

`import fmiopendata`

`from fmiopendata.wfs import download_stored_query`
`lightning2 = download_stored_query("fmi::observations::lightning::simple")`

`print(lightning2.latitudes)  # Latitude of the lightning event [° North]`
`print(lightning2.longitudes)  # Longitude of the lightning event [° East]`
`print(lightning2.times)  # Time of the lightning event [datetime]`
`print(lightning2.cloud_indicator)  # Indicator for cloud flashes (1 == cloud lightning)`
`print(lightning2.multiplicity)  # Multiplicity of the lightning event`
`print(lightning2.peak_current)  # Maximum current of the lightning event [kA]`
`print(lightning2.ellipse_major)  # Location accuracy of the lightning event [km]`

The weird thing is, I have got the lightning1 script to work for a couple of times but have no idea why it sometimes works and sometimes not.

@pnuu

**Comment by pnuu** [comment]
Thanks for reporting this!

Looking at the actual FMI open data [query](http://opendata.fmi.fi/wfs?service=WFS&version=2.0.0&request=getFeature&storedquery_id=fmi::observations::lightning::multipointcoverage), the results are empty, meaning there are no lightning strikes available. If you try a time period with lightning, the query works, so it's just bad handling of empty results.

```python
 lightning1 = download_stored_query("fmi::observations::lightning::multipointcoverage",
                                                             args=["starttime=2020-07-06T18:00:00Z",
                                                                       "endtime=2020-07-06T20:00:00Z"])
```

I'll fix the handling of empty results for `*::multipointcoverage` later today after work.

**Comment by jeppoo1** [comment]
Thank you! This was actually a rather silly mistake for me... I should have guessed that weather data changes all the time and there may not be lighting data every day and every hour 😜

**Comment by pnuu** [comment]
This will be fixed by #8 . After the tests have passed, I'll create a new release and push the new version to PyPi.

**Comment by pnuu** [comment]
Version 0.3.2 now available on PyPi has these fixes. Upgrade by `pip install -U fmiopendata`.
